src/main/java/com/zaxxer/nuprocess/linux/LinuxProcess.java

```diff
@@ -141,37 +134,44 @@ private void prepareProcess(List<String> command, String[] environment, Path cwd
       // See https://github.com/JetBrains/jdk8u_jdk/blob/master/src/solaris/classes/java/lang/ProcessImpl.java#L86
       byte[] envBlock = toEnvironmentBlock(environment);
 
-      // See https://github.com/JetBrains/jdk8u_jdk/blob/master/src/solaris/classes/java/lang/ProcessImpl.java#L96
```

Comment:
I removed this comment because it's unrelated to the code it's attached to. The Java code now creates its own pipes, rather than using `{-1, -1, -1}` like it did when this comment was added.

I don't 100% understand why we can't take advantage of the pipe creation code in `forkAndExec`, but having tested going back to `{-1, -1, -1}` it seems like we can't. Most of the time it works, but randomly it seems to hang. Really bizarre. I can see that you [started with `{-1, -1, -1}`](https://github.com/brettwooldridge/NuProcess/commit/e838dbbba7a43be2458de4fc571cef62404ae0a6#diff-515a1963c8aa9b6ec932c230bed92969R90) but then [switched away to fix a "race condition seen once"](https://github.com/brettwooldridge/NuProcess/commit/dc9491e20e5b8db9170ac6e968bf9a1ebc48e78d#diff-515a1963c8aa9b6ec932c230bed92969R87), so it seems like you've already had that experience yourself. (I only tested switching back out of curiosity, just to be clear.)